zebra/irdp_main.c

```diff
@@ -234,7 +234,8 @@ int irdp_send_thread(struct thread *t_advert)
       }
 
   tmp = irdp->MaxAdvertInterval-irdp->MinAdvertInterval;
-  timer =  (random () % tmp ) + 1;
+  assert (tmp > 0);
+  timer = (random () % tmp) + 1;
   timer = irdp->MinAdvertInterval + timer;
```

Comment:
the RFC doesn't actually allow min == max:
```
      MinRtrAdvInterval
                     The minimum time allowed between sending
                     unsolicited multicast Router Advertisements from
                     the interface, in seconds.  MUST be no less than 3
                     seconds and no greater than .75 *
                     MaxRtrAdvInterval.

                     Default: 0.33 * MaxRtrAdvInterval If
                     MaxRtrAdvInterval >= 9 seconds; otherwise, the
                     Default is MaxRtrAdvInterval.
```
which is fixed by errata 3154 to say
```
Default: 0.33 * MaxRtrAdvInterval If MaxRtrAdvInterval >= 9 seconds;
otherwise, the Default is 0.75 * MaxRtrAdvInterval.
```

I would suggest we either ditch the restriction altogether and allow `min == max` or go by the RFC strictly and enforce the 75% rule, not do our own custom restriction ;)

(Either way min & max are worded in a way that sounds bounds-including to me, so `random % (tmp + 1)` is correct)